mito-ai/src/Extensions/AppPreview/StreamlitPreviewPlugin.tsx

```diff
@@ -76,6 +76,40 @@ export interface IStreamlitPreviewManager {
   getCurrentPreview(): MainAreaWidget | null;
 }
 
+/**
+ * Simple placeholder widget for loading state.
+ */
+class PlaceholderWidget extends Widget {
+  constructor() {
+    super();
+    this.addClass('jp-placeholder-widget');
+    
+    const container = document.createElement('div');
+    container.style.display = 'flex';
+    container.style.flexDirection = 'column';
+    container.style.alignItems = 'center';
+    container.style.justifyContent = 'center';
+    container.style.height = '100%';
+    container.style.width = '100%';
+    container.style.padding = '20px';
+    
+    const message = document.createElement('div');
+    message.textContent = 'Building your app...';
```

Comment:
Maybe we should say "Building your app. This might take a couple of minutes" or something like that?